storage/jellyfish-merkle/src/tree_cache/mod.rs

```diff
@@ -213,7 +213,7 @@ where
         let root_node_key = self.get_root_node_key();
         let root_hash = self
             .get_node(root_node_key)
-            .unwrap_or_else(|_| panic!("Root node with key {:?} must exist", root_node_key))
+            .unwrap_or_else(|_| unreachable!("Root node with key {:?} must exist", root_node_key))
```

Comment:
This should be genuinely unreachable, but I would need a little more scrutiny.

Reply:
By all means. An annotation such as unreachable! should always be an invitation to double check. After all, a human is making an assertion that something cannot actually happen. Sometimes humans are wrong.